flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1458,26 +1456,15 @@ private static AbstractInvokable loadAndInstantiateInvokable(
 	private static class TaskCanceler implements Runnable {
 
 		private final Logger logger;
+		private final Task task;
 		private final AbstractInvokable invokable;
 		private final Thread executer;
-		private final String taskName;
-		private final ResultPartition[] producedPartitions;
-		private final InputGate[] inputGates;
-
-		public TaskCanceler(
-				Logger logger,
-				AbstractInvokable invokable,
-				Thread executer,
-				String taskName,
-				ResultPartition[] producedPartitions,
-				InputGate[] inputGates) {
 
+		TaskCanceler(Logger logger, Task task, AbstractInvokable invokable, Thread executer) {
```

Comment:
Why are you introducing circular dependency here between `Task` and `TaskCanceler`? There are various reasons why this is bad, including: is it necessary to expose 27 public methods (including things like `startTaskThread()` or `run()`) to the `TaskCanceler`?

In various different places we are trying to get away from this pattern of passing `StreamingTask` everywhere.

Reply:
The motivation is for reusing the `closeOrFailNetworkResources` and avoid static method and pass the arrays of `ResultPartition` and `InputGate` explicitly.  The previous `AbstractInvokable` could also be replaced and got from new `Task` parameter.

I think the previous introduced `AbstractInvokable` here is also not a good way considering exposing more public methods besides `AbstractInvokable#cancel()`. Comparing with `Task` parameter, we might add more public methods to do so. I agree with reverting this change to pass the previous specific three parameters here even though the `closeOrFailNetworkResources` might seem ugly.